File: app/src/main/java/it/liceoarzignano/bold/home/HomeCardBuilder.java
package it.liceoarzignano.bold.home;

import java.util.ArrayList;
import java.util.List;

public class HomeCardBuilder {
    private final List<String> mTitleList = new ArrayList<>();
    private final List<String> mContentList = new ArrayList<>();
    private int mCounter;
    private String mName;
    private HomeCard.HomeCardClickListener mClickListener;

    public HomeCardBuilder setName(String name) {
        this.mName = name;
        return this;
    }

    public HomeCardBuilder addEntry(String title, String content) {
        if (mCounter > 2) {
            return this;
        }

        mCounter++;
        mTitleList.add(title);
        mContentList.add(content);
        return this;
    }

    public HomeCardBuilder setOnClick(HomeCard.HomeCardClickListener listener) {
        mClickListener = listener;
        return this;
    }

    public HomeCard build() {
        return new HomeCard(mCounter, mName, mTitleList, mContentList, mClickListener);
    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/home/HomeCardBuilder.kt
package it.liceoarzignano.bold.home

import java.util.*

class HomeCardBuilder {
    private val mTitleList = ArrayList<String>()
    private val mContentList = ArrayList<String>()
    private var mCounter: Int = 0
    private var mName = ""
    private var mClickListener: HomeCard.HomeCardClickListener? = null

    fun setName(name: String): HomeCardBuilder {
        this.mName = name
        return this
    }

    fun addEntry(title: String, content: String): HomeCardBuilder {
        if (mCounter > 2) {
            return this
        }

        mCounter++
        mTitleList.add(title)
        mContentList.add(content)
        return this
    }

    fun setOnClick(listener: HomeCard.HomeCardClickListener): HomeCardBuilder {
        mClickListener = listener
        return this
    }

    fun build(): HomeCard = HomeCard(mCounter, mName, mTitleList, mContentList, mClickListener)
}
